vatesfr/xen-orchestra
# feat(mixins/SslCertificate): Let's Encrypt support

## Test recipe

- create a xo, listenning to port 80 and 443
- expose this server to the internet
- [ ] ssl certificate should be broken ( missing/self signed)
- add certDomain = 'domainname.xxx' to the https part of the config
- reload the server
- load the https page
- [ ] it should have a valid certificate after a few seconds 
- [ ] next loading should not re compute a certificate 
- [ ] accessing XO through another domain (localhost) should show a self signed certificate
- [ ] accessing XO through the IP shoul show a self seigned certifiate
- [ ] redirect to https shouldn't redirect challenge
 
### Check list

> Check if done, if not relevant leave unchecked.

- [ ] PR reference the relevant issue (e.g. `Fixes #007` or `See xoa-support#42`)
- [ ] if UI changes, a screenshot has been added to the PR
- `CHANGELOG.unreleased.md`:
  - [x] enhancement/bug fix entry added
  - [x] list of packages to release updated (`${name} v${new version}`)
- **I have tested added/updated features** (and impacted code)
  - [ ] unit tests (e.g. [`cron/parse.spec.js`](https://github.com/vatesfr/xen-orchestra/blob/b24400b21de1ebafa1099c56bac1de5c988d9202/%40xen-orchestra/cron/src/parse.spec.js))
  - [ ] if `xo-server` API changes, the corresponding test has been added to/updated on [`xo-server-test`](https://github.com/vatesfr/xen-orchestra/tree/master/packages/xo-server-test)
  - [x] at least manual testing

### Process

1. create a PR as soon as possible
1. mark it as `WiP:` (Work in Progress) if not ready to be merged
1. when you want a review, add a reviewer (and only one)
1. if necessary, update your PR, and re- add a reviewer

From [_the Four Agreements_](https://en.wikipedia.org/wiki/Don_Miguel_Ruiz#The_Four_Agreements):

1. Be impeccable with your word.
1. Don't take anything personally.
1. Don't make assumptions.
1. Always do your best.
